modules/friendlytag.js

```diff
@@ -1438,76 +1467,97 @@ Twinkle.tag.callback.evaluate = function friendlytagCallbackEvaluate(e) {
 		case 'article':
 			params.tags = form.getChecked( 'articleTags' );
 			params.group = form.group.checked;
-			params.tagParameters = {
-				cleanup: form["articleTags.cleanup"] ? form["articleTags.cleanup"].value : null,
-				closeParaphrasing: form["articleTags.closeParaphrasing"] ? form["articleTags.closeParaphrasing"].value : null,
-				copyEdit: form["articleTags.copyEdit"] ? form["articleTags.copyEdit"].value : null,
-				copypaste: form["articleTags.copypaste"] ? form["articleTags.copypaste"].value : null,
-				expandLanguageLangCode: form["articleTags.expandLanguageLangCode"] ? form["articleTags.expandLanguageLangCode"].value : null,
-				expandLanguageArticle: form["articleTags.expandLanguageArticle"] ? form["articleTags.expandLanguageArticle"].value : null,
-				globalize: form["articleTags.globalize"] ? form["articleTags.globalize"].value : null,
-				notability: form["articleTags.notability"] ? form["articleTags.notability"].value : null
-			};
-			// {{expert needed}} parameters:
-			params.expertNeeded = form["articleTags.expertNeeded"] ? form["articleTags.expertNeeded"].value : null,
-			params.expertNeededTalk = form["articleTags.expertNeededTalk"] ? form["articleTags.expertNeededTalk"].value : null,
-			params.expertNeededReason = form["articleTags.expertNeededReason"] ? form["articleTags.expertNeededReason"].value : null,
-			// common to {{merge}}, {{merge from}}, {{merge to}}
-			params.mergeTarget = form["articleTags.mergeTarget"] ? form["articleTags.mergeTarget"].value : null;
-			params.mergeReason = form["articleTags.mergeReason"] ? form["articleTags.mergeReason"].value : null;
-			params.mergeTagOther = form["articleTags.mergeTagOther"] ? form["articleTags.mergeTagOther"].checked : false;
-			// common to {{not English}}, {{rough translation}}
-			params.translationLanguage = form["articleTags.translationLanguage"] ? form["articleTags.translationLanguage"].value : null;
-			params.translationNotify = form["articleTags.translationNotify"] ? form["articleTags.translationNotify"].checked : null;
-			params.translationPostAtPNT = form["articleTags.translationPostAtPNT"] ? form["articleTags.translationPostAtPNT"].checked : null;
-			params.translationComments = form["articleTags.translationComments"] ? form["articleTags.translationComments"].value : null;
+
+			// Save values of input text and checkbox parameters, this works as
+			// Quickform input fields within subgroups of elements with name 'articleTags'
+			// have their name attribute as 'articleTags.' + name of the subgroup element
+			$(form).find('input[type=text]').each(function(idx,el) {
+				// el are the HTMLInputElements, el.name gives their name attribute
+				params[el.name.slice(12)] = form[el.name].value;
+			});
+			$(form).find("input[type=checkbox][name^='articleTags.']").each(function(idx,el) {
```

Comment:
Nifty though this is, `select` dropdown menus are left out, breaking `notability`, `globalize`, and `Should be PNG`.

Reply:
Dealt with. Textareas covered as well.